gymnasium/spaces/box.py

```diff
@@ -211,13 +211,14 @@ def sample(self, mask: None = None) -> NDArray[Any]:
 
         sample[upp_bounded] = (
             -self.np_random.exponential(size=upp_bounded[upp_bounded].shape)
-            + self.high[upp_bounded]
+            + high[upp_bounded]
         )
 
         sample[bounded] = self.np_random.uniform(
             low=self.low[bounded], high=high[bounded], size=bounded[bounded].shape
         )
-        if self.dtype.kind == "i":
+
+        if self.dtype.kind != "f":
```

Comment:
Does this effect integer types as well? Why we only applying this to floats now not integers

Reply:
Yes, previously `floor` was applied only to integers. Now it is applied to everything that is not float (bools, ints and units). I may change this to be more explicit. This fixes bools not being `floor`ed and having only `True` values in the output (because anything other than `0` would be casted to `True` and when we have a continuous variable it is very unlikely for it to be `0`).